cirq/google/serializable_gate_set.py

```diff
@@ -98,15 +111,32 @@ def serialize(self,
             msg = v2.program_pb2.Program()
         msg.language.gate_set = self.gate_set_name
         if isinstance(program, circuits.Circuit):
-            self._serialize_circuit(program, msg.circuit)
+            self._serialize_circuit(program,
+                                    msg.circuit,
+                                    arg_function_language=arg_function_language)
+            if arg_function_language is None:
+                arg_function_language = (
+                    arg_func_langs._infer_function_language_from_circuit(
```

Comment:
Iterating through a large circuit could be a decent sized overhead.  Why can't we just assume the most permissive in this case?  Isn't that why you added None to the dictionary above?

Reply:
We want to use the least permissive, in order to run against older servers.